src/buildExec.ts

```diff
@@ -46,9 +64,20 @@ const buildExec = () => {
   const slug = core.getInput('slug');
   const swift = core.getInput('swift');
   const swiftProject = core.getInput('swift_project');
-  const token = core.getInput('token');
-  const upstream = core.getInput('upstream_proxy');
   const url = core.getInput('url');
+  let token = core.getInput('token');
+  if (token === '') {
+    let codecovURL = url;
+    if (codecovURL === '') {
+      codecovURL = 'https://codecov.io';
+    }
+    try {
+      token = await core.getIDToken(codecovURL);
+    } catch (error) {
+      core.debug(`Got error while retrieving id token: ${error}`);
```

Comment:
if original token is empty, and we cannot fetch id token for some reason (like no permissions in GHA pipeline) -> we will just skip the error and continue using without token. I think this is one case how people are using this action